infrastructure/src/main/java/org/corfudb/infrastructure/CompactorService.java

```diff
@@ -148,7 +151,12 @@ private void runOrchestrator() {
 
                 if (isLeader) {
                     if (managerStatus != null && managerStatus.getStatus() == StatusType.STARTED) {
-                        getCompactorLeaderServices().validateLiveness();
+                        if (distributedCheckpointerHelper.isCompactionDisabled()) {
```

Comment:
If we move this `if (isLeader)` block up to L143 (before `if (managerStatus != null)`), then it only takes one runOrchestrator run to mark cycle as FAILED/COMPLETED and shutdown the checkpoint JVM if necessary.

Reply:
Oh that's a good point. Will update.